plugin/sdkv2/debugging: Add VS Code and clarify Delve CLI usage (#2100)
hashicorp/terraform-website: content/plugin/sdkv2/debugging.mdx

### Starting A Provider In Debug Mode

> Once a provider has a debug mode added to its `main` function, it can be activated. Run your debugger, which is typically built into an editor or the Delve CLI, by passing it the provider binary as the command to execute. Specify any flags, environment variables, or other input that is necessary to start your provider.

Reviewer: Just trying to make this be more in active voice - please tweak as necessary if I've misunderstood the meaning! 

```suggestion
After you add a debug mode to your provider's `main` function, you can activate the debugger, which is typically built into an editor or the Delve CLI. To run your debugger, pass it the provider binary as the command to execute. Specify flags, environment variables, and any other input your provider needs to run.
```

#### Visual Studio Code

> Visual Studio Code (VS Code) features native [debugging](https://code.visualstudio.com/docs/editor/debugging) support with the [Go extension](https://marketplace.visualstudio.com/items?itemName=golang.go). The necessary launch configuration can be created within the repository as a `.vscode/launch.json` file, [user settings](https://code.visualstudio.com/docs/editor/debugging#_global-launch-configuration), or [workspace settings](https://code.visualstudio.com/docs/editor/multi-root-workspaces#_workspace-launch-configurations).

Reviewer: ```suggestion
Visual Studio Code (VS Code) features native [debugging](https://code.visualstudio.com/docs/editor/debugging) support with the [Go extension](https://marketplace.visualstudio.com/items?itemName=golang.go). You can create the necessary launch configuration within the repository as a `.vscode/launch.json` file, [user settings](https://code.visualstudio.com/docs/editor/debugging#_global-launch-configuration), or [workspace settings](https://code.visualstudio.com/docs/editor/multi-root-workspaces#_workspace-launch-configurations).
```

An example `.vscode/launch.json` configuration file:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug Terraform Provider",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            // this assumes your workspace is the root of the repo
            "program": "${workspaceFolder}",
            "env": {},
            "args": [
                "-debug",
            ]
        }
    ]
}
```

Running the editor debugger will output the reattach configuration to the debug console.

#### Delve CLI

Start a [`delve`](https://github.com/go-delve/delve) debugging session:

```sh
$ dlv exec --accept-multiclient --continue --headless ./terraform-provider-example -- -debug
```

It will print output like the following to `stdout`:

```
Provider started, to attach Terraform set the TF_REATTACH_PROVIDERS env var:

        TF_REATTACH_PROVIDERS='{"registry.terraform.io/my-org/my-provider":{"Protocol":"grpc","Pid":3382870,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin713096927"}}}'
```

> Connect your debugger, such as your editor or another delve CLI process, to the debugger server.

Reviewer: will folks know how to do this?

Author: Aye -- for editors it'll be product specific and we certainly do not have the bandwidth to investigate and document them all. 😅  Since this is the Delve CLI section, should probably show an example of that though. 👍 

``````suggestion
Connect your debugger, such as your editor or another delve CLI process, to the debugger server.

For example with Delve CLI, using the server address and port output by `dlv exec` such as `API server listening at: 127.0.0.1:55413`:

```shell
dlv connect 127.0.0.1:55413
```
``````
